crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -600,6 +600,19 @@ impl<'db> TypeInferenceBuilder<'db> {
         }
     }
 
+    /// Check if a given AST note is reachable.
```

Comment:
```suggestion
    /// Check if a given AST node is reachable.
```

Reply:
Also worth noting here that it doesn't work for arbitrary nodes? The node reachability has to have explicitly been recorded in semantic indexing.